fix(react-rich-text-renderer): Compile to commonJS

endymion1818 opened the pull request:
## Benefits:
We can use this package outside the Webiny ecosystem (eg. for using the HeadlessCMS to build a
NextJS application)

## Changes
Closes #2261

## How Has This Been Tested?
I tested the build with a NextJS application and it works.

## Documentation
I'll add this to the Changelog

Pavel910 commented on packages/react-rich-text-renderer/.babelrc.js:
```diff
@@ -1 +1,8 @@
+
+module.exports = {
+    ...defaults,
+    plugins: [...defaults.plugins, "@babel/plugin-transform-modules-commonjs"]
```
@endymion1818 This babel plugin dependency needs to be added to the `package.json` for that respective package.

endymion1818 committed:
fix(react-rich-text-renderer): compile to commonjs

We can use this package outside the Webiny ecosystem (eg. for using the HeadlessCMS to build a
NextJS application)

Pavel910 commented on packages/react-rich-text-renderer/package.json:
```diff
@@ -9,6 +9,7 @@
   "author": "Webiny Ltd",
   "license": "MIT",
   "dependencies": {
+    "@babel/plugin-transform-modules-commonjs": "^7.16.8",
```
@endymion1818 this should be in `devDependencies` because it's only relevant for us, at publish time. Once the package is published, users don't need this in their projects. It's not a dependency for the core package logic.

endymion1818 committed:
chore(react-rich-text-renderer): move to dev dependencies

Moved to dev dependencies as package is not needed for core package logic.
